### PR TITLE
[release/0.1 backport] remove containerd as dependency

### PR DESCRIPTION
**Pull request by thaJeztah**
- backport of https://github.com/containerd/nri/pull/51
- backport of https://github.com/containerd/nri/pull/61
- relates to https://github.com/containerd/containerd/pull/9107

(cherry picked from commit da8a7e53614c60c106f824caf3d120ea582a0cbe)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,14 +3,6 @@ module github.com/containerd/nri
 go 1.14
 
 require (
-	// when updating containerd, adjust the replace rules accordingly
-	github.com/containerd/containerd v1.5.0-beta.3
+	github.com/opencontainers/runtime-spec v1.0.3-0.20200929063507-e6143ca7d51d
```
Comment (thaJeztah) [inline go.mod:6]: This is the version that was used by containerd v1.5.0-beta.3; https://github.com/containerd/containerd/blob/v1.5.0-beta.3/go.mod#L45C2-L45C77

It's 12 commits ahead of runtime-spec v1.0.2; https://github.com/opencontainers/runtime-spec/compare/v1.0.2...e6143ca7d51d

We _could_ consider using a tagged release here 🤔

**Review by klihub — APPROVED** [review]
If we decide to go ahead with [#9107](https://github.com/containerd/containerd/pull/9107) for the 1.6 branch in containerd then I think this could be the way to do it.

**Comment by thaJeztah** [comment]
Yes, I definitely think we should consider doing this; it reduces dependency hell if this dependency is somewhere in the tree (and is responsible for the circular dependency). At least I don't think it _hurts_ to have it.

I would still love to see a tagged version for the runtime-spec dependency in it as well (https://github.com/containerd/nri/pull/58#discussion_r1335863210), but that's slightly more tricky as we'd either downgrade to v1.0.2 (perhaps that's fine as minimum version); or upgrade to v1.1.0, but that's a larger change; https://github.com/opencontainers/runtime-spec/compare/e6143ca7d51d...v1.1.0

**Review by samuelkarp — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (mikebrow) [reply go.mod:6]: nod to using tagged if possible

**Review by mikebrow — COMMENTED** [review]
just a nit a question on keeping to tagged if possible..

**Comment by thaJeztah** [comment]
fixed the typo (backported https://github.com/containerd/nri/pull/61), and I downgraded the minimum required version of runtime-spec to v1.0.2 (latest tagged version before v1.1.0). containerd 1.6 already depends on a more recent version of the runtime-spec (but uses a non-tagged version).


@mikebrow @samuelkarp PTAL

**Comment by thaJeztah** [comment]
If this is merged, we can consider tagging a v0.1.1, and I will update my PR in containerd; https://github.com/containerd/containerd/pull/9107

**Review by samuelkarp — APPROVED** [review]

**Review by mikebrow — APPROVED** [review]
LGTM

**Pull request merged**